runner/runner.py

```diff
@@ -0,0 +1,47 @@
+import typing
+from utils import utils
+from typing_extensions import Protocol
+
+
+class ScannerProtocol(Protocol):
+    headers = []  # type: typing.List[str]
```

Comment:
If you haven't seen this before, this is from [PEP 544](https://www.python.org/dev/peps/pep-0544/). Allows type checkers to check static duck typing. It's standard in Python 3.7, backported to 3.5/3.6 through the `typing-extensions` package.